### PR TITLE
adding vxlan to packet headers

### PR DESCRIPTION
**Pull request by chestercheng**
adding vxlan to packet headers

### CODE REVIEW

**Comment by ajbalogh** [comment]
@chestercheng thanks for your first time contribution. 

At this time we do not have a default PR template for the organization but one of the PR requirements would be validation. In this case we would be looking for a test case that validates the implementation you are proposing.  

The following test case https://github.com/open-traffic-generator/snappi-ixnetwork/blob/main/tests/vlan/test_vlan_fields.py is an example of validating packet header fields.

**Comment by chestercheng** [comment]
@ajbalogh no problem, i will add the test case later.

File: snappi_ixnetwork/trafficitem.py [hunk 1]
```diff
@@ -298,6 +300,15 @@ class TrafficItem(CustomField):
         "order": ["length", "data"],
     }
 
+    _VXLAN = {
+        "flags": "vxlan.header.flags",
+        "reserved0": "vxlan.header.reserved",
+        "vni": "vxlan.header.vni",
+        "reserved1": "vxlan.header.reserved8",
+        "order": ["flags", "reserved0", "vni", "reserved1"],
+        "convert_int_to_hex": ["flags"],
```
Comment (alakendu) [inline snappi_ixnetwork/trafficitem.py:309]: Looks like "reserved0" and "reserved1" also required "convert_int_to_hex"

File: tests/vxlan/test_vxlan_fields.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    outer_eth.dst.value = outer_dst_mac
+    outer_eth.ether_type.value = ether_type
+
+    ip.src.value = src_ip
+    ip.dst.value = dst_ip
+
+    udp.src_port.value = src_port
+    udp.dst_port.value = dst_port
+
+    vxlan.flags.value = flags
```
Comment (alakendu) [inline tests/vxlan/test_vxlan_fields.py:40]: Configure and validation for "reserved0" and "reserved1"

File: tests/vxlan/test_vxlan_fields.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    Configure three raw vxlan flows with ,
+    - fixed pattern for all vxlan fields
+    - list pattern for all vxlan fields
+    - counter pattern for all vxlan fields
+    Validate,
+    - fetch the vxlan header via restpy framework and validate
+      against expected.
+    """
+    # fixed
+    outer_src_mac = "00:00:0a:00:00:01"
```
Comment (alakendu) [inline tests/vxlan/test_vxlan_fields.py:12]: We also validate packet through "e2e" test where validate it in capture level. Please check if we can incorporate additional UT with "@pytest.mark.e2e" marker. Please check this test for more details https://github.com/open-traffic-generator/snappi-ixnetwork/blob/main/tests/udp/test_udp_fixed_ports_e2e.py

**Review by alakendu — APPROVED** [review]

File: snappi_ixnetwork/trafficitem.py (hunk 1 above)
Comment (chestercheng) [reply snappi_ixnetwork/trafficitem.py:309]: thanks for the correction

File: tests/vxlan/test_vxlan_fields.py (hunk 3 above)
Comment (chestercheng) [reply tests/vxlan/test_vxlan_fields.py:12]: @alakendu okay, I will do it.

**Comment by ashutshkumr** [comment]
@chestercheng could you please sync with latest of `main` in order to automatically run CI on PR (incoming from forked repo) - this was previously not supported.

**Comment by chestercheng** [comment]
> @chestercheng could you please sync with latest of `main` in order to automatically run CI on PR (incoming from forked repo) - this was previously not supported.

no problem

**Pull request merged**